File: Data Structures & Algorithms/palindromic-substrings/submission-1.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int n=s.size();
        int ans=0;
        for(int i=0;i<n;i++){
            int k=0;
            while(i-k>=0 && i+k<n && s[i-k]==s[i+k]){
                ans++;
                k++;
            }
            int l=i,r=i+1;
            while(l>=0 && r<n && s[l]==s[r]){
                ans++;
                l--;
                r++;
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/anagram-groups/submission-1.js
class Solution {
    /**
     * @param {string[]} strs
     * @return {string[][]}
     */
    groupAnagrams(strs) {
        const mp={};
        for(let str of strs){
            const dp=new Array(26).fill(0);
            const a='a'.charCodeAt(0);
            for(let ch of str){
                dp[ch.charCodeAt(0)-a]++;
            }
            let key='';
            for(let i of dp){
                key+=String(i);
            }
            if(!mp[key]){
                mp[key]=[str];
            }else{
                mp[key].push(str);
            }
        }
        const ans=[];
        for(let key in mp){
            ans.push(mp[key]);
        }
        return ans;
    }
}
